pkg/signer/kmssigner.go

```diff
@@ -34,31 +34,33 @@ type KMSSigner struct {
 	client kmsiface.KMSAPI
 	// keyID is the KMS Key Id used for signing
 	keyID string
+	// public key
+	publicKey crypto.PublicKey
 }
 
 // New returns a KMSSigner instance given and AWS client and a KMS key used for signing.
 // TODO: explain what are the pre-requisits for the KMS key.
-func New(client kmsiface.KMSAPI, keyID string) *KMSSigner {
-	return &KMSSigner{
-		client: client,
-		keyID:  keyID,
-	}
-}
-
-// Public returns the public key corresponding to the opaque, private key.
-// TODO: Do we really need this method? Error handling is inhexistant. Maybe another function?
-func (s *KMSSigner) Public() crypto.PublicKey {
-	response, err := s.client.GetPublicKey(&kms.GetPublicKeyInput{
```

Comment:
Will moving this into `signer.New` cause issues as it isn't fetched every time?

Reply:
it shouldn't since the public key is kept inside instantiated structure